From 54ab16247ff1be333bdf757a324ee02a660e4742 Mon Sep 17 00:00:00 2001
From: Justus Winter <justus@sequoia-pgp.org>
Date: Tue, 20 Feb 2024 23:13:07 +0100
Subject: [PATCH] gpg: Use u32 as discriminant in enum Error.

  - Fixes building on 32 bit architectures, where the new
    GPG_ERR_MINUS_ONE was not representable.

  - Fixes b2ecab6cbaf6f834db5ff52f34d8ce550decd124.

  - Fixes #41.
---
 src/Makefile        | 1 +
 src/error_codes.inc | 1 +
 2 files changed, 2 insertions(+)

diff --git a/src/Makefile b/src/Makefile
index 0eee2b1..8f34868 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -28,6 +28,7 @@ error_codes.inc: err-codes.h.in
 	echo >"$@.new" "/// Error codes defined in libgpg-error."
 	echo >>"$@.new" "#[derive(Clone, Copy, Debug, PartialEq, Eq)]"
 	echo >>"$@.new" "#[allow(non_camel_case_types)]"
+	echo >>"$@.new" "#[repr(u32)]"
 	echo >>"$@.new" "pub enum Error {"
 	sed -n -e '/[0-9]\+\s\+GPG_ERR_.*\s\+[A-Z].*/p' "$<" | while read N SYMBOL DESC; do \
 	    echo >>"$@.new" "    /// $$DESC."; \
diff --git a/src/error_codes.inc b/src/error_codes.inc
index c60558e..1f3fa42 100644
--- a/src/error_codes.inc
+++ b/src/error_codes.inc
@@ -1,6 +1,7 @@
 /// Error codes defined in libgpg-error.
 #[derive(Clone, Copy, Debug, PartialEq, Eq)]
 #[allow(non_camel_case_types)]
+#[repr(u32)]
 pub enum Error {
     /// Success.
     GPG_ERR_NO_ERROR = 0,
-- 
GitLab

